### PR TITLE
Handle duplicate CONNECT frame

### LINKED ISSUE #179: Gracefully handle clients that attempt to "reauthenticate"
See Grigory Starkin's findings on rabbitmq-users [1][2][3].

Currently the plugin assumes that a connection cannot "reauthenticate". In practice it indeed won't but apparently some clients allow arbitrary frames to be sent at any time. False positives around peer
unavailability detection and lousy client state management in libraries can also result in "double authentication".

We should handle this more gracefully. At the very least there should be no scary looking connection process crashes in the logs. We can consider ignoring reauthentication attempts (for security reasons)
and logging a warning or log an error and close the connection. The first option seems to be more user friendly and error communication back to clients in MQTT leaves a lot to be desired, of course 🤦‍♂️.

1. https://groups.google.com/d/msg/rabbitmq-users/1MBdymMAZzw/HQLDwI_8GAAJ
2. https://groups.google.com/d/msg/rabbitmq-users/1MBdymMAZzw/hCmbM7Y1FgAJ
3. https://groups.google.com/d/msg/rabbitmq-users/1MBdymMAZzw/0ch8vfX-FwAJ

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #179

Uses undefined MQTT return code 255 to mean "duplicate auth attempt"

### CODE REVIEW

File: include/rabbit_mqtt_frame.hrl [hunk 1]
```diff
@@ -36,11 +36,12 @@
...
--define(CONNACK_INVALID_ID,  2). %% identifier rejected
--define(CONNACK_SERVER,      3). %% server unavailable
--define(CONNACK_CREDENTIALS, 4). %% bad user name or password
--define(CONNACK_AUTH,        5). %% not authorized
+-define(CONNACK_PROTO_VER,   1).   %% unacceptable protocol version
+-define(CONNACK_INVALID_ID,  2).   %% identifier rejected
+-define(CONNACK_SERVER,      3).   %% server unavailable
+-define(CONNACK_CREDENTIALS, 4).   %% bad user name or password
+-define(CONNACK_AUTH,        5).   %% not authorized
+-define(CONNACK_DUP_AUTH,    255). %% duplicate authentication attempt
```
Comment (lukebakken) [inline include/rabbit_mqtt_frame.hrl]: We can use a different value, I just picked the last one in the "reserved for future use" range:

http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/csd02/mqtt-v3.1.1-csd02.html#_Toc385349257

Comment (michaelklishin) [reply include/rabbit_mqtt_frame.hrl]: I think that would trip up some clients. I'm afraid if we are being forgiving in this case we should use a `CONNACK_ACCEPT` 🤷‍♂️

Comment (lukebakken) [reply include/rabbit_mqtt_frame.hrl]: I tried that first. If we do that, new channels and what-not are opened by the processor. I'll see if that can be skipped.

Comment (lukebakken) [reply include/rabbit_mqtt_frame.hrl]: Yeah, that won't be a problem.

**Comment by michaelklishin** [comment]
Backported to `v3.7.x`.

**Pull request merged**